www/apps/book/app/basics/modules/page.mdx

```diff
@@ -0,0 +1,272 @@
+
+A module is integrated as a building block in your Medusa application, without implications on the existing setup.
+
+<Note title="Tip">
+
+Modules and services have other uses too, which you'll learn about in later chapters.
+
+</Note>
+
+---
+
+## How to Create a Module?
+
+Modules are created in a sub-directory of `src/modules`.
+
+For example, create the directory `src/modules/hello`.
+
+### 1. Create Data Model
+
+A data model represents a table in the database. It's created in a TypeScript or JavaScript file under the module's `models` directory.
+
+For example, create the file `src/modules/hello/models/my-custom.ts` with the following content:
+
+```ts title="src/modules/hello/models/my-custom.ts"
+import { model } from "@medusajs/framework/utils"
+
+const MyCustom = model.define("my_custom", {
+  id: model.id().primaryKey(),
+  name: model.text(),
+})
+
+export default MyCustom
+```
+
+You define the data model using the `define` method of the `model` utility imported from `@medusajs/framework/utils`. It accepts two parameters:
+
+1. The first one is the name of the data model's table in the database. It should be snake-case.
+2. The second is an object, which is the data model's schema. The schema's properties are defined using the `model`'s methods.
+
+The example above defines the data model `MyCustom` with the properties `id` and `name`.
+
+<Note>
+
+Data models automatically have the date properties `created_at`, `updated_at`, and `deleted_at`.
+
+</Note>
+
+### 2. Create Service
+
+A module must define a service that implements its functionalities, such as manage the records of your custom data models in the database.
+
+A service is a TypeScript or JavaScript class defined in the `service.ts` file at the root of your module's directory.
+
+For example, create the file `src/modules/hello/service.ts` with the following content:
+
+export const highlights = [
+  ["4", "MedusaService", "The service factory function."],
+  ["5", "MyCustom", "The data models to generate data-management methods for."]
+]
+
+```ts title="src/modules/hello/service.ts" highlights={highlights}
+import { MedusaService } from "@medusajs/framework/utils"
+import MyCustom from "./models/my-custom"
+
+class HelloModuleService extends MedusaService({
+  MyCustom,
+}){
+}
+
+export default HelloModuleService
+```
+
+Since the module has data models, its service extends a class generated by the service factory function `MedusaService` imported from `@medusajs/framework/utils`.
+
+<Note title="Tip">
+
+If a module doesn't have data models, it doesn't need to extend `MedusaService`.
+
+</Note>
+
+The service factory function accepts as a parameter an object of data models. It returns a service class with data management methods, such as `createMyCustoms` or `retrieveMyCustoms`.
+
+Your service extends the returned class to benefit from the generated methods.
```

Comment:
suggestion:

In the snippet above, your module service extends `MedusaService`. `MedusaService` can take a set of data models and will generate a class that has implementations for the CRUD operations on those data models. For example, the `HelloModuleService` now allows you to do `createMyCustoms` or `retrieveMyCustoms`.

Using `MedusaService` is not required, but it is helpful and recommended when your module has data models.